instrumentation/apache-pulsar/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/info/ClientEnhanceInfo.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pulsar.info;
+
+import io.opentelemetry.instrumentation.api.field.VirtualField;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Producer;
+
+/**
+ * for producer and consumer, cache service_url and determine whether the instance has been
+ * enhanced.
+ */
+public class ClientEnhanceInfo {
+  public final String topic;
+  public final String brokerUrl;
+
+  public ClientEnhanceInfo(String topic, String brokerUrl) {
+    this.topic = topic;
+    this.brokerUrl = brokerUrl;
+  }
+
+  public static void virtualField(Producer<?> instance, ClientEnhanceInfo info) {
```

Comment:
In my opinion `virtualField` is not a good method name here as it does not convey what this method does.